test/e2e/enveloped_object_placement_test.go

```diff
@@ -0,0 +1,309 @@
+			*placedEnvelopeWebhook.Webhooks[0].FailurePolicy != *testEnvelopeWebhook.Webhooks[0].FailurePolicy ||
+			*placedEnvelopeWebhook.Webhooks[0].SideEffects != *testEnvelopeWebhook.Webhooks[0].SideEffects ||
+			*placedEnvelopeWebhook.Webhooks[0].MatchPolicy != *testEnvelopeWebhook.Webhooks[0].MatchPolicy {
+			return fmt.Errorf("webhook config does not match")
+		}
+		if len(placedEnvelopeWebhook.Webhooks[0].Rules) != 1 {
+			return fmt.Errorf("webhook rule size does not match")
+		}
+		if diff := cmp.Diff(placedEnvelopeWebhook.Webhooks[0].Rules[0],
+			testEnvelopeWebhook.Webhooks[0].Rules[0],
+			cmpopts.IgnoreFields(admv1.Rule{}, "Scope")); diff != "" {
+			return fmt.Errorf("webhook rule diff (-got, +want): %s", diff)
+		}
+		return nil
+	}
+}
+
+func checkForOneClusterFailedToApplyStatus(wantSelectedResources []placementv1beta1.ResourceIdentifier) func() error {
+	crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+	workNamespaceName := fmt.Sprintf(workNamespaceNameTemplate, GinkgoParallelProcess())
+	failedResourcePlacement := []placementv1beta1.FailedResourcePlacement{
+		{
+			ResourceIdentifier: placementv1beta1.ResourceIdentifier{
+				Kind:      "ResourceQuota",
+				Name:      testEnvelopeResourceQuota.Name,
+				Version:   "v1",
+				Namespace: testEnvelopeResourceQuota.Namespace,
+				Envelope: &placementv1beta1.EnvelopeIdentifier{
+					Name:      testEnvelopConfigMap.Name,
+					Namespace: workNamespaceName,
+					Type:      placementv1beta1.ConfigMapEnvelopeType,
+				},
+			},
+			Condition: metav1.Condition{
+				Type:   string(placementv1beta1.WorkConditionTypeApplied),
+				Status: metav1.ConditionFalse,
+				Reason: work.AppliedManifestFailedReason,
+			},
+		},
+	}
+
+	return func() error {
+		crp := &placementv1beta1.ClusterResourcePlacement{}
+		if err := hubClient.Get(ctx, types.NamespacedName{Name: crpName}, crp); err != nil {
+			return err
+		}
+		wantCRPConditions := crpSyncFailedConditions(crp.Generation)
+		if diff := cmp.Diff(crp.Status.Conditions, wantCRPConditions, crpStatusCmpOptions...); diff != "" {
+			return fmt.Errorf("CRP status diff (-got, +want): %s", diff)
+		}
+		// check the selected resources is still right
+		if diff := cmp.Diff(crp.Status.SelectedResources, wantSelectedResources, crpStatusCmpOptions...); diff != "" {
+			return fmt.Errorf("CRP status diff (-got, +want): %s", diff)
+		}
+		// check the placement status has a failed placement
+		applyFailed := false
+		for _, placementStatus := range crp.Status.PlacementStatuses {
+			if len(placementStatus.FailedResourcePlacements) != 0 {
+				applyFailed = true
+			}
+		}
+		if !applyFailed {
+			return fmt.Errorf("CRP status does not have failed placement")
+		}
+		for _, placementStatus := range crp.Status.PlacementStatuses {
+			// this is the cluster that got the new enveloped resource that was malformed
+			if len(placementStatus.FailedResourcePlacements) != 0 {
```

Comment:
Once the state is stable, you can compare the whole CRP status similar to https://github.com/Azure/fleet/blob/main/test/e2e/placement_selecting_resources_test.go#L802.

The readability will be better.

Reply:
there is no guarantee which cluster has the failed resource placement